src/test/java/pl/coderslab/imageManager/services/ImageServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package pl.coderslab.imageManager.services;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import pl.coderslab.imageManager.model.ImageDescription;
+import pl.coderslab.imageManager.repositories.ImageRepository;
+
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReader;
+import javax.imageio.metadata.IIOMetadata;
+import javax.imageio.stream.ImageInputStream;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+
+class ImageServiceTest {
+
+    private static final Logger log = LoggerFactory.getLogger(ImageServiceTest.class);
+    private ImageService imageService;
+    private ImageRepository imageRepository;
+
+    Path path = Paths.get("src/main/webapp/images/0/fruits.jpeg");
+    File file = path.toFile();
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    @BeforeEach
+    public void setUp() {
+        imageRepository = mock(ImageRepository.class);
+        imageService = new ImageService(imageRepository);
+    }
+
+    @Test
+    public void when_saving_image_return_image() {
```

Comment:
Hmm, ten test jest zły z powodów jakie opisałem poniżej plus jest zdecydowanie zbyt skomplikowany. Każdy test powinien mieć wydzielone trzy sekcje, czyli:

given:
// inicjalizuje jakieś obiekty

when:
// tu jest zwykle 1 linijka - wywołanie tego co chcę testować

then:
// sprawdzenie, czyli robimy asercje

To czego mi tutaj brakuje, to właśnie tych 3 sekcji i czytelności, plus dodatkowo jak pisałem poniżej działamy na mockach, więc wszystkie operacje bazodanowe nie działają